### ISSUE TITLE
wrong cci calculation

### ISSUE THREAD

**Issue by angopinath** [bukosabino/ta#63]
for cci calculation, we should use "mean absolute deviation (mad)" not standard deviation. please correct it. 
unfortunately, pandas rolling doesn't provide mad function. we have to find it by our own.

here I'm posting my code which is used for cci calculation.

```
pp = (data['CLOSE'] + data['LOW'] + data['HIGH'] )/3
mad = lambda x : np.mean(np.abs(x-np.mean(x)))
data['CCI']= (pp-pp.rolling(period).mean())/(0.015 * pp.rolling(period).apply(mad,True))
```

**Comment by bukosabino** [comment]
Hi @angopinath ,

Thank you for your comment. Yes, I am wrong. 

I have checked your mad function and I am not sure. Whey mean(mean())? Probably we should use mean(std()). What do you think?

```python
mad = lambda x : np.mean(np.abs(x-np.mean(x)))
or 
mad = lambda x : np.mean(np.abs(x-np.std(x)))
```

**Comment by MrTrai** [comment]
Any resolution to this?

**Comment by alik604** [comment]
Current: 
`pp = (high + low + close) / 3.0
    cci = (pp - pp.rolling(n, min_periods=0).mean()) / (c * pp.rolling(n, min_periods=0).std())
    if fillna:
        cci = cci.replace([np.inf, -np.inf], np.nan).fillna(0)
    return pd.Series(cci, name='cci')`

Based on [another library](https://github.com/FreddieWitherden/ta/blob/master/ta.py#L114) and [wikipedia](https://en.wikipedia.org/wiki/Commodity_channel_index#Calculation)
I think `cci = (pp - pp.rolling(n, min_periods=0).mean()) / (c * pp.rolling(n, min_periods=0).mean())` might be better. 
On the contary [investopedia](https://www.investopedia.com/terms/c/commoditychannelindex.asp) agrees with the original implemenation.  
I am not familiar with the practical differences between tho two. I'd suggest a (optional) boolean flag to be used, so the programmer decide, I think this is a good middle ground. 

If you wish, I would be happy to make the change a create a pull (FYI it would be my first)

**Comment by juliangall** [comment]
To match the formula on Investopedia, this is the correct code, so far as I can see.

`pp = (data['CLOSE'] + data['LOW'] + data['HIGH'] )/3  
(pp-pp.rolling(20).mean()) / (0.15 * abs(pp-pp.rolling(20).mean()).rolling(20).mean())`

The denominator takes the absolute values of the last (period) differences and averages them, which is what the Investopedia article refers to as the "mean deviation".

**Comment by bukosabino** [comment]
I think this is the correct implementation. 

```python
    pp = (high + low + close) / 3.0
    mad = lambda x : np.mean(np.abs(x-np.mean(x)))
    cci = ((pp - pp.rolling(n, min_periods=0).mean())
           / (c * pp.rolling(n, min_periods=0).apply(mad, True)))
```

It is included in the new ta version `0.4.6`.

Thank you to all of you.

**Comment by traderblakeq** [comment]
Remember: Tradingview's default settings in CCI indicator is calculated on Close not hlc3.

- So if you get different numbers... correct the settings on TV.
